state-state.js

```diff
@@ -43,4 +45,54 @@ module.exports = function StateState() {
 		getParent: getParent,
 		getParentName: getParentName
 	}
+
```

Comment:
At that point, it's sort of back to the shared utility file.

It might feel more natural if the properties were added elsewhere, like an extension. I.E. If all the code that was just added to state-state.js was moved to a new file.

Something like:

```js
var StateState = require('./state-state')
var stateExtensions = require('./state-extensions')
//...
var prototypalStateHolder = StateState()
stateExtensions(prototypalStateHolder)
```

I'm not sure if that would make it cleaner or not.